veloxms-core/src/main/scala/edu/berkeley/veloxms/models/Model.scala

```diff
@@ -72,30 +78,62 @@ abstract class Model[T: ClassTag] extends Logging {
    * @param nextVersion
    * @return
    */
-  def retrainFeatureModelsInSpark(observations: RDD[(UserID, T, Double)], nextVersion: Version): RDD[(T, FeatureVector)]
+  def retrainFeatureModelsInSpark(
+      observations: RDD[(UserID, T, Double)],
+      nextVersion: Version): RDD[(T, FeatureVector)]
+
+  final def retrainUserWeightsInSpark(
+      itemFeatures: RDD[(T, FeatureVector)],
+      observations: RDD[(UserID, T, Double)],
+      nextVersion: Version)
+    : RDD[(UserID, WeightVector)] = {
+
+    val newAvgUser = retrainAvgUserInSpark(itemFeatures, observations)
+    averageUser.put(newAvgUser, nextVersion)
 
-  final def retrainUserWeightsInSpark(itemFeatures: RDD[(T, FeatureVector)], observations: RDD[(UserID, T, Double)]): RDD[(UserID, WeightVector)] = {
     val obs = observations.map(x => (x._2, (x._1, x._3)))
     val ratings = itemFeatures.join(obs).map(x => (x._2._2._1, x._2._1, x._2._2._2))
     UserWeightUpdateMethods.calculateMultiUserWeights(ratings, numFeatures)
   }
 
   /**
-   * Velox implemented - fetch from local Tachyon partition
+   * Calculates the optimal cold-start user model for the case when we have
+   * no training data for a user.
+   *
+   * Implementation: Take the average score for each item across the
+   * entire training data set and use that as the training data for
+   * the average user model.
```

Comment:
I'm wondering if this is really what we want to be doing. This may give more weight to users with more observations. Still, great to have something initial in place!